tfhe/src/core_crypto/experimental/entities/lwe_shrinking_keyswitch_key.rs

```diff
@@ -0,0 +1,328 @@
+//! Module containing the definition of the [`LweShrinkingKeyswitchKey`].
+
+use crate::core_crypto::commons::parameters::*;
+use crate::core_crypto::commons::traits::*;
+use crate::core_crypto::entities::*;
+use crate::core_crypto::experimental::commons::parameters::{
+    LweSecretKeySharedCoefCount, LweSecretKeyUnsharedCoefCount,
+};
+
+/// An [`LWE shrinking keyswitch key`](`LweShrinkingKeyswitchKey`) is an [`LWE keyswitch
+/// key`](`LweKeyswitchKey`) where the input and output key share a part of their randomness.
+///
+/// See [`the formal definition of an LWE keyswitch key`](`LweKeyswitchKey#formal-definition`).
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub struct LweShrinkingKeyswitchKey<C: Container>
```

Comment:
If the output_key can have unshared coeffs, the keyswitch can increase the dimension
Then the name `Shrinking` can be confusing

Reply:
now that we force the keys to be shared we keep the shrinking name